src/Entity/OgMembership.php

```diff
@@ -319,6 +319,25 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException(sprintf('Entity type %s with ID %s is not an OG group.', $entity_type_id, $group->id()));
     }
 
+    // Check for an existing membership.
+    if (!$this->isNew()) {
+
+      $query = \Drupal::entityQuery('og_membership');
+      $count = $query
+        ->condition('uid', $this->get('uid')->target_id)
+        ->condition('entity_id', $this->get('entity_id')->value)
+        ->condition('entity_type', $this->get('entity_type')->value)
+        // Filter out this membership.
+        ->condition('id', $this->id(), '<>')
```

Comment:
@dakala fyi, here's a better solution - we do a `COUNT` query, and make sure to filter out the `$this->id()` (it was missing in your previous logic)